Modify rule S3358: LaYC for Nested ternaries (#2106)
SonarSource/rspec: rules/S3358/cfamily/rule.adoc

== Why is this an issue?

:noncompliant: cfamily/noncompliant.adoc

:compliant: cfamily/compliant.adoc

include::../description.adoc[]

> === Exceptions

Reviewer: Maybe rework this paragraph as well with a concrete example? For example:
"
In C++11, nested ternaries inside constexpr functions are compliant. Indeed, in C++11, such functions are limited to a single return statement, so no other solution is possible. The later standards lifted this restriction, so this exception is no longer valid.
+ code example
"

Author: Done

== Why is this an issue?

:noncompliant: cfamily/noncompliant.adoc

:compliant: cfamily/compliant.adoc

include::../description.adoc[]

=== Exceptions

> In {cpp}11 mode the issue is not raised for ternary operators inside `constexpr` functions.

Reviewer: ```suggestion
In {cpp}11, the rule ignores ternary operators inside `constexpr` functions.
```
